### DISCUSSION TITLE
[Chapter 02]

### DISCUSSION THREAD

**Discussion by coalong** [cowaine/spring-boot-for-msa#10] [Chapter 02. Framework and Spring Boot] [answered]
1. Spring 의 3가지 핵심 요소 중 AOP 에 대한 직접 코드 실습을 하면서, 굉장히 신기 했었던 기억이 있다.
    Spring AOP 동작은 어떤 방식으로 구현 했을까?

2. 팀프로젝트를 진행할 때, 프로젝트 규모가 조금씩 커지다보니 여러 라이브러리를 사용할 때가 있었다.
    그 과정에서 라이브러리 버전 충돌 문제, 라이브러리간 버전 호환 문제 때문에 꽤나 고생했는데,
    이런 버전 문제를 잘 관리할 수 있는 방법이 있을까?

**Comment by corock** [comment, marked as answer]
1. AOP 에 대해 자세히 알아볼 수 있는 7장에서 동작 원리까지 이참에 정리하면 좋을 것 같아요 😄
2. 저는 스프링 프레임워크에선 BOM 을, 스프링 부트 프레임워크를 사용하는 경우 starter 의존성을 적극 사용하려고 합니다.
